### ISSUE TITLE
JsonSerializationException in DocumentSession.Query<T>(string sql)

### ISSUE THREAD

**Issue by Gikerl** [JasperFx/marten#1389]
Hello,

I am trying to use the query my database with an SQL string as the input parameter of the Query<T>() method. However, a Newtonsoft.Json.JsonSerializationException is being thrown. Doing the same query with without the SQL and with a Where statement, the exception isn't thrown.

I'm using Marten 3.10.0, PostgreSQL 10.10 for Ubuntu, .Net Framework 4.62

Am I using the SQL String Query wrong?

Example Unit Test:

``` csharp
[TestFixture]
public class QueryTests
{
    private static IDocumentSession CreateSession()
        => DocumentStore.For(ConfigureStoreOptions).LightweightSession();

    private static void ConfigureStoreOptions(StoreOptions storeOptions)
    {
        storeOptions.Schema.For<IUser>().AddSubClassHierarchy();
        storeOptions.AutoCreateSchemaObjects = AutoCreate.All;

        storeOptions.Schema.For<IUser>().GinIndexJsonData();

        storeOptions.Connection("host=localhost;database=MartenDB;password=password;username=someuser");
    }

    [Test]
    public void QueryWithSQLString_ReturnUser()
    {
        // Test setup
        var instance = CreateSession();


        var id = Guid.NewGuid();
        var user = new User(id, "SomeName", "LastName", id.ToString());
        using (var session = CreateSession())
        {
            session.Store(user);
            session.SaveChanges();
        }


        // Exercise system
        var result = instance.Query<IUser>($"Where id = \'{id}\'");


        // Verify outcome
        Assert.That(result.Count, Is.EqualTo(1));
        Assert.That(result[0], Is.Not.Null);


        // Teardown
    }

    [Test]
    public void QueryWithLinq_ReturnUser()
    {
        // Test setup
        var instance = CreateSession();


        var id = Guid.NewGuid();
        var user = new User(id, "SomeName", "LastName", id.ToString());
        using (var session = CreateSession())
        {
            session.Store(user);
            session.SaveChanges();
        }


        // Exercise system
        var result = instance.Query<IUser>().Where(u => u.Id == id).ToList();


        // Verify outcome
        Assert.That(result.Count, Is.EqualTo(1));
        Assert.That(result[0], Is.Not.Null);


        // Teardown
    }
}
```

**Comment by oskardudycz** [comment]
@Gikerl I apologize for the late feedback. It should be fixed with https://github.com/JasperFx/marten/pull/1479 and available in v3.12

**Comment by jeremydmiller** [comment]
Fixed by #1479
